src/BloomExe/web/EnhancedImageServer.cs

```diff
@@ -1,51 +1,137 @@
-﻿// Copyright (c) 2014 SIL International
+﻿// Copyright (c) 2014-2015 SIL International
 // This software is licensed under the MIT License (http://opensource.org/licenses/MIT)
 using System;
 using System.Diagnostics;
-using System.Drawing.Text;
 using System.Globalization;
-using System.Linq;
-using System.Net;
-using Bloom.ImageProcessing;
 using System.IO;
-using L10NSharp;
+using System.Threading;
 using Microsoft.Win32;
+using L10NSharp;
 using Palaso.IO;
 using Bloom.Collection;
-
+using Bloom.ImageProcessing;
 
 namespace Bloom.web
 {
 	/// <summary>
 	/// A local http server that can serve (low-res) images plus other files.
 	/// </summary>
+	/// <remarks>geckofx makes concurrent requests of URLs which this class handles. This means
+	/// that the methods of this class get called on different threads, so it has to be
+	/// thread-safe.</remarks>
 	public class EnhancedImageServer: ImageServer
 	{
 		private FileSystemWatcher _sampleTextsWatcher;
 		private bool _sampleTextsChanged = true;
+		private const int RWLockTimeOut = 100;
```

Comment:
I was wondering whether '100' was measured in milliseconds. I suppose when I first saw it I was also wondering whether that would be enough to e.g. read a large image from disk, but since this lock is only used for simple property access it seems very generous.